plugins/analytics/public/threatminer.py

```diff
@@ -0,0 +1,724 @@
+import logging
+import requests
+import json
+from core.analytics import OneShotAnalytics
+from core.observables import Hash, Ip, Hostname, Url
+from core.config.config import yeti_config
+from core.errors import ObservableValidationError
+from hashlib import sha1
+
+
+class ThreatMinerApi(object):
+
+    API_URL = "https://api.threatminer.org/v2/"
+
+    @staticmethod
+    def fetch(observable, params, uri):
+        try:
+            url = ThreatMinerApi.API_URL + uri
+            response = requests.get(
+                url, params=params, proxies=yeti_config.proxy)
+            if response.ok:
+                return response.json()
+        except Exception as e:
+            logging.warning("Hit an error checking {},{}".format(
+                observable.value, e
+                ))
+
+
+"""
+    Pulls metadata related to a file hash.
+    This consists of various hash types and file properties including filename.
+    However it looks like file name is sometimes a url.
+"""
+
+
+class MetaData(OneShotAnalytics, ThreatMinerApi):
+    default_values = {
+        "group":
+            "ThreatMiner",
+        "name":
+            "Retrieve metadata.",
+        "description":
+            "Checks for any meta data stored in ThreatMiner."
+    }
+
+    ACTS_ON = ["Hash"]
+
+    @staticmethod
+    def analyze(observable, results):
+        links = set()
+        params = {"q": observable.value, "rt": 1}
+        json_result = ThreatMinerApi.fetch(observable, params, "sample.php")
+        if json_result:
+            json_string = json.dumps(
+                json_result,
+                sort_keys=True,
+                indent=4,
+                separators=(",", ": ")
+            )
+
+            result = dict(
+                [
+                    ("raw", json_string),
+                    ("source", "threatminer_query")
+                ]
+            )
+            results.update(raw=json_string)
+            _results = json_result.get("results")
+
+            if not _results:
+                return list(links)
+
+            for r in _results:
+                hashes = {
+                    "md5": r["md5"],
+                    "sha1": r["sha1"],
+                    "sha256": r["sha256"]
+                    }
+
+                create_hashes = [
+                    (k, v) for k, v in hashes.items() if v != observable.value
+                ]
+
+                for k, v in create_hashes:
```

Comment:
It's probably better to iterate over `hashes.items()` here and check that the hash is new inside that loop

Something like:

```python
for family, _hash in hashes.items():
  if _hash == observable.value:
    continue
  try:
    new_hash = ...
```